core/templates/pages/exploration-editor-page/modal-templates/welcome-modal.component.spec.ts

```diff
@@ -0,0 +1,98 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for WelcomeModalComponent.
+ */
+
+import { Component, NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, waitForAsync, TestBed } from '@angular/core/testing';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { ContextService } from 'services/context.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { WelcomeModalComponent } from './welcome-modal.component';
+
+@Component({
+  selector: 'oppia-changes-in-human-readable-form',
+  template: ''
+})
+class ChangesInHumanReadableFormComponentStub {
+}
+
+class MockActiveModal {
+  close(): void {
+    return;
+  }
+
+  dismiss(): void {
+    return;
+  }
+}
+
+describe('Welcome Modal Component', () => {
+  let component: WelcomeModalComponent;
+  let fixture: ComponentFixture<WelcomeModalComponent>;
+  let ngbActiveModal: NgbActiveModal;
+  let contextService: ContextService;
+  let siteAnalyticsService: SiteAnalyticsService;
+  const explorationId = 'exp1';
+  let siteAnalyticsServiceSpy: jasmine.Spy;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        WelcomeModalComponent,
+        ChangesInHumanReadableFormComponentStub
+      ],
+      providers: [
+        ContextService,
+        SiteAnalyticsService,
+        UrlInterpolationService, {
+          provide: NgbActiveModal,
+          useClass: MockActiveModal
+        },
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(WelcomeModalComponent);
+    component = fixture.componentInstance;
+
+    ngbActiveModal = TestBed.inject(NgbActiveModal);
+    contextService = TestBed.inject(ContextService);
+    siteAnalyticsService = TestBed.inject(SiteAnalyticsService);
+    siteAnalyticsServiceSpy = spyOn(
+      siteAnalyticsService, 'registerTutorialModalOpenEvent');
+    spyOn(contextService, 'getExplorationId').and.returnValue(explorationId);
+    fixture.detectChanges();
+  });
+
+  it ('should evaluate exploration id when component is initialized', () => {
+    expect(component.explorationId).toBe(explorationId);
+    expect(component.editorWelcomeImgUrl).toBe(
+      '/assets/images/general/editor_welcome.svg');
+    expect(siteAnalyticsServiceSpy)
+      .toHaveBeenCalled();
+  });
+
+  it ('should close the modal', () => {
+    const dismissSpy = spyOn(ngbActiveModal, 'dismiss').and.callThrough();
+    component.cancel();
+    expect(dismissSpy).toHaveBeenCalledWith();
```

Comment:
```suggestion
    const dismissSpy = spyOn(ngbActiveModal, 'dismiss').and.callThrough();

    component.cancel();

    expect(dismissSpy).toHaveBeenCalledWith();
```

Reply:
I think that is harder to read because all the test setup, runs, and expects are wrapped inside a describe block. I've checked the other spec files, and I don't really see this type of style. Should we get more eyes on this?